### PR TITLE
[DSMR-Binding] Support for DateTimeFormat "yy-MM-dd HH:mm:ss" used by ACE4000 GTMM

### PR DESCRIPTION
**Pull request by Truxen**
As the CloudBees Build-Server is not working: is someone able to compile the DSMR-Binding for me?
I have currently no IDE ;)

### CODE REVIEW

File: bundles/binding/org.openhab.binding.dsmr/src/main/java/org/openhab/binding/dsmr/internal/cosem/CosemDate.java [hunk 1]
```diff
@@ -75,8 +75,22 @@ protected DateTimeType parse(String cosemValue) throws ParseException {
 
 			return new DateTimeType(c);
 		} else {
-			throw new ParseException("value: " + cosemValue
-					+ " is not a valid CosemDate string (yyMMddHHmmss)", 0);
+			 p = Pattern.compile("(\d{2}-\d{2}-\d{2}\s\d{2}:\d{2}:\d{2})([S,W]?)");
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.dsmr/src/main/java/org/openhab/binding/dsmr/internal/cosem/CosemDate.java]: please extract as constant

File: bundles/binding/org.openhab.binding.dsmr/src/main/java/org/openhab/binding/dsmr/internal/cosem/CosemDate.java [hunk 2]
```diff
@@ -75,8 +75,22 @@ protected DateTimeType parse(String cosemValue) throws ParseException {
...
+
+			m = p.matcher(cosemValue);
+			if (m.matches()) {
+				SimpleDateFormat formatter = new SimpleDateFormat("yy-MM-dd HH:mm:ss");
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.dsmr/src/main/java/org/openhab/binding/dsmr/internal/cosem/CosemDate.java]: please extract as constant

**Comment by teichsta** [comment]
Hi @Truxen thanks for this PR. @mvolaart could you please have a look at this one? Thanks in advance, Thomas E.-E.

**Comment by teichsta** [comment]
any news @Truxen @mvolaart?

**Comment by mvolaart** [comment]
I had already contact with @Truxen.

The issue is that the DSMR-specification has in it's lifetime different datetime specifications. Beside this the exact format is (especially in older versions) not strictly specified, leaving it to the interpretation of the meter supplier.
Because of this situation i don't like the 'trial-error'  pattern of this patch.

It solves the issue certainly for the ACE4000. In my development version i have a different patch but i am not satisfied with that patch either.

So we can merge the patch for the short term, but need a better solution for the long term (I'm working on this).

Marcel

**Comment by teichsta** [comment]
superseded by #2926

Nevertheless thanks @Truxen for contributing to openHAB!

**Pull request closed without merging**